pkg/client/query/cache/config.go

```diff
@@ -0,0 +1,69 @@
+package cache
+
+import (
+	"time"
+)
+
+// EvictionPolicy determines how items are removed when number of keys in the cache reaches MaxKeys.
+type EvictionPolicy int64
+
+const (
+	FirstInFirstOut = EvictionPolicy(iota)
+	LeastRecentlyUsed
+	LeastFrequentlyUsed
+)
+
+// queryCacheConfig is the configuration for query caches. It is intended to be
+// configured via QueryCacheOptionFn functions.
+type queryCacheConfig struct {
+	// MaxKeys is the maximum number of items (key/value pairs) the cache can
+	// hold before it starts evicting.
+	MaxKeys        int64
+	EvictionPolicy EvictionPolicy
+	// TTL is how long items should remain in the cache. Items older than the TTL
+	// MAY not be evicted but SHOULD not be considered as cache hits.
```

Comment:
> What determines if it will or will not be evicted? MAY NOT is a bit confusing.

The implementation determines that, this config doesn't know anything about the implementation.
In the case of `InMemoryCache`, eviction only happens when on a per-key/value basis, when a key/value is updated.

> Should we have a WILL NOT be considered cache hits? I can't tell what the behaviour is with SHOULD NOT

:+1: Agreed.